### PR TITLE
Issue 35284: Hitting enter when configuring a product to add to order…

### LINKED ISSUE #35284: Magento Admin - Hitting enter when configuring a product to add to order in admin does not save the custom options
### Preconditions (*)

1. Magento ver. 2.4.2-p2
2.

### Steps to reproduce (*)

In Magento 2, navigate to Sales > Order and create a new order

Select "Add Products" to add a product to the order from the product list.

In the product list, find an upholstery product and select "Configure". In the configure pop-up, select the custom options and enter the quantity. From the quantity field HIT ENTER instead of using the cursor to select "Ok" in the top right corner. Note that the configuration pop up window has closed.

add the product to the order and return to the order page. Scroll to the product you just added and note that you see an error "The product has required options". Select "Configure" on the product and note that the custom options that were set before are no longer present, suggesting they were not saved.

Once you reconfigure the options, the error goes away. On the order page, if you use "Enter" rather than selecting "Ok" in the configuration pop-up, the configurations are saved as appropriate.

Current behavior (please include any screenshots or videos as needed):

When I select different custom options for a MTO product and then hit "enter" on the keyboard to apply changes and close the window, the custom options are not saved. When I add the product to the order, and find the product, I notice that I receive an error "The product has required options" and I am forced to reconfigure the product options

Here's a loom video showing what I did to encounter the error:  [https://www.loom.com/share/40dab8667bef40a882c513f52eb2985a]

**Expected behavior:** 

I should be able to hit "Enter" to apply the custom options and close the window and expect that the options are saved when I add the product to the order.

**Additional Information:**

2.4-develop branch behavior: Product got selected but "The requested qty is not available" error is displayed on clicking "Add selected products" button. 
This issue is not reproducible for Bundle Products on 2.4-develop branch.On hitting "Enter" instead of OK button, product is not getting selected.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ **X** ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by rogerdz**
### Description (*)

Magento Admin - Hitting enter when configuring a product to add to order in admin does not save the custom options

### Fixed Issues (if relevant)

1. Fixes magento/magento2#35284

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogerdz** [comment]
@magento run all tests

**Review by edenduong — APPROVED** [review]
Thank you for your contribution !
P/S: This PR can be covered by MFTF test (It is similar `AdminCreateOrderWithConfigurableProductTest` test case with `AdminAddToOrderConfigurableProductActionGroup`  action group). You can created it to cover this change. Could you please add it ? I think it is more better.

**Review by edenduong — CHANGES_REQUESTED** [review]
Thank you for your contribution !
P/S: This PR can be covered by MFTF test (It is similar AdminCreateOrderWithConfigurableProductTest test case and AdminAddToOrderConfigurableProductActionGroup action group with small modify). You can created new test case to cover this change. Could you please add it ? I think it is more better.

**Comment by rogerdz** [comment]
@edenduong I added MFTF test

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminCreateOrderWithConfigurableProductPressKeyEnterTest">
+        <annotations>
+            <title value="Create Order in Admin with configurable product"/>
+            <stories value="AC-2040"/>
+            <description value="Create order with configurable product."/>
+            <features value="Sales"/>
+            <testCaseId value="AC-2040"/>
```
Comment (edenduong) [inline app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml]: It is duplicate testCaseId in Magento. I think you can leave it as empty like that test case

image

**Review by edenduong — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml (hunk 1 above)
Comment (rogerdz) [reply app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml]: I removed it ^^

Comment (rogerdz) [reply app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml]: I added new commit

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml [hunk 2]
```diff
@@ -9,6 +9,13 @@
 <tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
     <test name="AdminCreateOrderWithConfigurableProductPressKeyEnterTest">
+        <annotations>
+            <title value="Create Order in Admin with configurable product"/>
+            <description value="Create order with configurable product."/>
```
Comment (edenduong) [inline app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml]: Could you please add "with pressing enter key in option select modal" to title and description to separate it with the normal flow ? Thanks

**Review by edenduong — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml (hunk 2 above)
Comment (rogerdz) [reply app/code/Magento/Sales/Test/Mftf/Test/AdminCreateOrderWithConfigurableProductPressKeyEnterTest.xml]: thank @edenduong, I added new commit

**Comment by rogerdz** [comment]
@magento run all tests

**Review by edenduong — APPROVED** [review]

**Comment by engcom-Lima** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Lima** [comment]
This PR is already merged. Hence, moving this PR to recently merged.

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**